### PR TITLE
added tests around minified runtime versions of semver functions

### PR DESCRIPTION
**Pull request by mpeyper**
In #11271 I wrote some tests to help try an work out what was happening with the runtime versions of the semver functions.  I offered to add them to the test suite in that PR and @sokra suggested as a seperate PR to not hold up the bug fix getting through.

There's some pretty nasty code in here to generate the functions with a dummy `RuntimeTemplate` and evaluate the functions to be runnable in the test suite, so I'm happy to take suggestions on how to make those parts less nasty.  I did investigate using `new Function(...)` instead of `eval`, but there was an issue with accessing other functions in the module's scope when executing it in a NodeJS environment ([some details in the MSDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function#Difference_between_Function_constructor_and_function_declaration)).

**What kind of change does this PR introduce?**

Tests only.

**Did you add tests for your changes?**

I only added tests.

**Does this PR introduce a breaking change?**

No.

**What needs to be documented once your changes are merged?**

Nothing.

### CODE REVIEW

File: test/SemVer.unittest.js [hunk 1]
```diff
@@ -57,6 +72,64 @@ describe("SemVer", () => {
 		]);
 	});
 
+	it("should parseVersion correctly (runtime)", () => {
+		const parseVersionRuntime = createRuntimeFunction(parseVersionRuntimeCode);
```
Comment (sokra) [inline test/SemVer.unittest.js:76]: Could you avoid duplicating the test by using e.g.

```js
for (const [name, fn] of [["normal", parseVersion], ["runtime", parseVersionRuntime]])
```

Comment (mpeyper) [reply test/SemVer.unittest.js:76]: Could do. I considered it but thought the tests that are already quite loopy would get harder to follow what's going on.

Happy to update if you prefer.

Comment (mpeyper) [reply test/SemVer.unittest.js]: Done

**Comment by sokra** [comment]
Thanks

**Pull request merged**